Rubberduck.Parsing/ComReflection/ComProject.cs

```diff
@@ -44,21 +45,15 @@ public class ComProject : ComBase
         private readonly List<ComStruct> _structs = new List<ComStruct>();
         public IEnumerable<ComStruct> Structs => _structs;
 
-        public IEnumerable<IComType> Members
-        {
-            get
-            {
-                //Note - Enums and Types should enumerate *last*. That will prevent a duplicate module in the unlikely(?)
-                //instance where the TypeLib defines a module named "Enums" or "Types".
-                return _modules.Cast<IComType>()
-                    .Union(_interfaces)
-                    .Union(_classes)
-                    .Union(_enumerations)
-                    .Union(_structs);
-            }
-        } 
+        //Note - Enums and Types should enumerate *last*. That will prevent a duplicate module in the unlikely(?)
+        //instance where the TypeLib defines a module named "Enums" or "Types".
```

Comment:
Isn't that an indication of a smelly design? If it falls over flat just because we encountered a module named `Enum`, then something's wrong, no? We should be able to tell an `enum` apart from a module named `enum` simply by looking at the type, no?

Reply:
Probably lazy design more than anything - building that assumption into the interface makes [this section of code](https://github.com/rubberduck-vba/Rubberduck/blob/next/Rubberduck.Parsing/ComReflection/ReferencedDeclarationsCollector.cs#L144) much, much more efficient and terse. The _root_ of that problem (IIR) was that there isn't an `EnumerationDeclaration` - it just uses the generic one because the only thing that differs is the `DeclarationType` .  The assumption has pretty much always been there (even prior to the refactor that split them out I think).  #LetSleepingBearsLie.